Add tests for models

EmilMingaleev opened the pull request.

EmilMingaleev committed:
Add styles and code for gems

EmilMingaleev committed:
Add styles and code for gems

EmilMingaleev committed:
code for gems

EmilMingaleev committed:
Add tests for models using rspec, capybara, factory_bot, faker, database_cleaner

EmilMingaleev committed:
lust update

DenisZackharov commented on app/assets/stylesheets/styles.css:
```diff
@@ -4,7 +4,7 @@ body {
     padding: 0;
     font-family: 'Roboto', sans-serif;
-    background-image: url('background.jpg');
+    background-image: url('1.jpg');
```
Это зачем ты решил добавить?)

DenisZackharov commented on app/views/articles/edit.html.erb:
```diff
@@ -1,5 +1,3 @@
-
-
 <h1><%= @article.title %></h1>
```
лучше с первой строчки писать, поправь пожалуйста

DenisZackharov replied:
поправь пожалуйста

DenisZackharov commented on Gemfile.lock:
а как ты добавил гем shoulda-matchers?

Похоже на то что ты его на комп поставил через `gem install shoulda-matchers` это не совсем правильно ведь если я себе склонирую твой проект то у меня этого гема не будет

Добавлять гемы нужно всегда в gemfile потом запускать `bundle install` чтобы обновить gemfile.lock

Таким образом я склонировав проект запущу bundle install и все гемы для проекта легко поставлю

DenisZackharov commented on spec/models/article_spec.rb:
```diff
@@ -0,0 +1,17 @@
+require "rails_helper"
+RSpec.describe Article, type: :model do
```
Оставь просто так, require 'rails_helper' не обязательно
```
describe Article, type: :model do
```

DenisZackharov commented on spec/models/article_spec.rb:
```diff
@@ -0,0 +1,17 @@
+require "rails_helper"
+RSpec.describe Article, type: :model do
+  it "is not valid without a title" do
```
Тесты лучше называть так что они будут возвращать, например
it "returns invalid article"

А уже ситуации отделять с помощью context
```
context "without status" do
  let(:status) { nil }

  it "returns invalid article" do
      article = Article.new(title: "Test Title", body: "Test Content", status: status)
      expect(article).not_to be_valid
    end
  end
```

DenisZackharov commented on spec/spec_helper.rb:
```diff
@@ -1,3 +1,4 @@
+require "capybara/rspec"
```
Можешь убрать пока, так как ты его пока не используешь, даже гем не добавил еще

DenisZackharov replied:
Сори гем вижу, выпили его пока capybara

DenisZackharov commented on app/controllers/comments_controller.rb:
```diff
@@ -1,22 +1,25 @@
+  def comment_params
+    params.require(:comment).permit(:commenter, :body, :status)
   end
-  
```
Этот минус значит что надо добавить пустую строку в конце, нужно это сделать везде

DenisZackharov commented on .rubocop.yml:
```diff
@@ -0,0 +1,25 @@
+ Enabled: false
+
+Style/StringLiterals:
+  EnforcedStyle: double_quotes
```
Странно что везде строки поменяны с двойной ковычки на одинарную, тут же написано что двойные юзаем, прогони пожалуйста bin/rubocop -a чтобы исправилось

DenisZackharov commented on db/migrate/20240413201323_add_users.rb:
нельзя просто взять и удалить миграцию бд, если хочешь удалить таблицу нужно писать новую миграцию по удалению таблицы, файл миграции нужно сохранить для историчности

И вообще зачем удалил?

DenisZackharov commented on app/models/user.rb:
зачем удалил?

EmilMingaleev committed:
some fix

DenisZackharov commented on Gemfile:
```diff
@@ -6,7 +6,7 @@ gem "brakeman"
 gem "rubocop", "~> 1.47.0"
 gem "rubocop-rails"
-
+gem "shoulda-matchers"
```
Смотри тут добавляют гем только для тестовой среды, лучше тоже так сделать потому что этот гем не нужен в остальных средах

https://github.com/thoughtbot/shoulda-matchers

DenisZackharov commented on app/controllers/articles_controller.rb:
```diff
@@ -53,4 +53,5 @@ def destroy
   def article_params
     params.require(:article).permit(:title, :body, :status)
   end
+  
```
не перед последним end а в самом конце файла нужна пустая строка

DenisZackharov commented on bin/quality:
а это ты откудова взял?

EmilMingaleev committed:
returns migration and fix some comments

DenisZackharov commented on .rubocop.yml:
```diff
@@ -0,0 +1,25 @@
+  EnforcedStyle: double_quotes
+
+Bundler/OrderedGems:
+  Enabled: true
```
Это я душу но лучше добавь пустую строчку, и перед отправкой мне сам такие моменты проверяй

DenisZackharov commented on app/assets/images/background.jpeg:
🤡

DenisZackharov commented on app/assets/stylesheets/styles.css:
```diff
@@ -0,0 +1,11 @@
+       background-position: center;
+  }
+  
+  
```
та же тема с последней строчкой, но тут вообще 3 строчки в конце а нужна 1

еще эти свойства уехали по табам куда-то

background-size: cover;
background-position: center;

DenisZackharov commented on app/views/layouts/application.html.erb:
```diff
@@ -6,6 +6,8 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+    <%= stylesheet_link_tag 'styles', media: 'all' %>
```
не нужно под каждый новый файл со стилями добавлять теги, лучше сделай 1 файл application.css и добавь css свойства туда, и не нужно будет здесь тег создавать

Позже если интересно я расскажу как можно это организовать еще лучше

DenisZackharov commented on app/assets/stylesheets/styles.css:
лучше стили писать в файле application.css, не создавая новые файлы, а вообще есть еще лучше практики но о них мы позже поговорим

DenisZackharov commented on bin/brakeman:
У тебя во всех файлах этой папки одинарные кавычки, запусти `bin/rubocop -a` чтобы оно автоисправилось

DenisZackharov commented on db/schema.rb:
```diff
@@ -10,32 +10,24 @@
+    t.index ['article_id'], name: 'index_comments_on_article_id'
   end
 
-  create_table "users", force: :cascade do |t|
```
ты вернул миграцию на создание пользователя, но не выполнил миграцию заново, чтобы вернуть users таблицу в схему сначала 

Откати все миграции bin/rails db:rollback STEP=10
Потом накати снова bin/rails db:migrate

DenisZackharov replied:
и почему тут везде стали одинарные кавычки, выполни rubocop

DenisZackharov commented on app/controllers/articles_controller.rb:
👍

DenisZackharov reviewed (changes requested):
поправь все замечания

DenisZackharov commented on app/views/comments/_form.html.erb:
в этой форме много лишних энтеров, лучше так сделать:

```
<%= form_with model: [ @article, @article.comments.build ] do |form| %>
  <p>
    <%= form.label :commenter %><br>
    <%= form.text_field :commenter %>
  </p>
  <p>
    <%= form.label :body %><br>
    <%= form.text_area :body %>
  </p>
  <p>
    <%= form.submit %>
  </p>
  <p>
    <%= form.label :status %><br>
    <%= form.select :status, Visible::VALID_STATUSES, selected: 'public' %>
  </p>
<% end %>
```

EmilMingaleev committed:
corrected all comments

DenisZackharov commented on .rubocop.yml:
```diff
@@ -0,0 +1,30 @@
+
+Bundler/OrderedGems:
+  Enabled: true
+  
```
Лишний пробел

DenisZackharov replied:
Всё еще есть

DenisZackharov commented on app/assets/stylesheets/application.css:
```diff
@@ -1,15 +1,10 @@
+    background-size: cover;
+    background-position: center;
+  }
+  
```
лишний пробел

EmilMingaleev committed:
- Changed .rubocop.yml
- Rolled out a new migration to remove reference
- Fixed tests
- Updated bin/quality
- Switched to Postgres

EmilMingaleev committed:
lust

DenisZackharov commented on .rspec:
```diff
@@ -0,0 +1,2 @@
+--require spec_helper
```
это можешь убрать так как spec_helper реквайрится в rails_helper

```suggestion
```

EmilMingaleev committed:
fix minuses

EmilMingaleev committed:
fix minuses

DenisZackharov reviewed (approved):
Хорошая работа
